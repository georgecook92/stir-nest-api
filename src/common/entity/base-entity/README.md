# BaseEntity

A typeorm base entity that provies a primary generated id column. You must call super with the desired entity prefix in the constructor.

### Example

```ts
import { BaseEntity } from './base-entity';

class MyEntity extends BaseEntity {
  constructor() {
    super('prefix_');
  }
}
```
