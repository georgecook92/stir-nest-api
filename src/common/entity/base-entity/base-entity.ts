import { BeforeInsert, PrimaryColumn } from 'typeorm';
import { Injectable } from '@nestjs/common';

import { generateId } from '../../utils/generate-id/generate-id';
import { EntityWithId } from '../interfaces/entity-with-id.interface';

const ENTITY_ID_LENGTH = 16;

@Injectable()
export abstract class BaseEntity implements EntityWithId {
  constructor(private readonly entityPrefix: string) {}

  @PrimaryColumn()
  public id: string;

  @BeforeInsert()
  public async generateID(): Promise<void> {
    this.id = await generateId(this.entityPrefix, ENTITY_ID_LENGTH);
  }
}
