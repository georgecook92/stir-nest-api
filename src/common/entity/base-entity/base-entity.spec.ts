import { BaseEntity } from './base-entity';

describe('BaseEntity', () => {
  class MockEntity extends BaseEntity {
    constructor() {
      super('test_');
    }
  }

  it('should attach the correct prefix to the entity', () => {
    const mockEntity = new MockEntity();
    expect(mockEntity).toMatchSnapshot();
  });
});
