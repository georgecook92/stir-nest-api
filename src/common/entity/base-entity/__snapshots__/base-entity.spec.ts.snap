// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseEntity should attach the correct prefix to the entity 1`] = `
MockEntity {
  "entityPrefix": "test_",
}
`;
